### ISSUE TITLE
Compilation error: FREETYPE_SOURCE

### ISSUE THREAD

**Issue by samiel60** [fontforge/fontforge#3487]
Compiling from source the last github version
(not yet official release) in LinuxMint

I'm trying to compile FF from source,
waiting for the new official released announced more or less three mounths ago.
However, the ./configure gives me the following error:

```
configure: error: 

The file "/home/samiel/src/github.com/fontforge/fontforge//src/truetype/ttobjs.h"
was not found: "/home/samiel/src/github.com/fontforge/fontforge/"
does not appear to be the top directory of the freetype
sources.

Please either set the environment variable FREETYPE_SOURCE
to the top directory of the freetype source tree, or
use the --disable-freetype-debugger option.
```
What really have I to do here?
Thank you

**Comment by skef** [comment]
The freetype debugger built from the source that `FREETYPE_SOURCE` points to is optional. If you:

1. Aren't building with the `--enable-freetype-debugger` or `--with-freetype-source` options, and
2. Don't have `FREETYPE_SOURCE` defined in the environment `configure` runs in

it's mysterious why configure would be running this test. So the first thing to do is make sure neither of those things are true and confirm that the source builds. 

Of course, if you *want* the freetype debugger you can download and unpack the the source (of a desired version) here: https://sourceforge.net/projects/freetype/files/freetype2/ and then unpack it and add the `--with-freetype-source=`[dir], where [dir] is the top level of the directory you unpack. (I'm going by the instructions here -- I haven't done this on my machine.) 

Anyway, if those options don't fix the problem update the bug and I'll try to work with you further.

**Comment by samiel60** [comment]
Without that option, the ./configure worls properly.
I downloaded freetype-2.9.1 source, unpack it in ~ and run:
` ./configure --with-freetype-source=/home/samiel/freetype-2.9.1/`
but I receive still
```
The file "/home/samiel/freetpe-2.9.1//src/truetype/ttobjs.h"
was not found: "/home/samiel/freetpe-2.9.1/"
does not appear to be the top directory of the freetype
source
``.

**Comment by skef** [comment]
Your error message has the string "freetpe" instead of "freetype".

**Comment by samiel60** [comment]
Oh, I was so stupid.... now working.
Al the end of make I further receive:
```
Summary of optional features:

			UseIt?	HaveIt?
  real (floating pt)		double
  programs		yes	yes
  native scripting	yes	yes
  python scripting	check	yes2.7
  python extension	check	yes
  freetype debugger		/home/samiel/freetype-2.9.1/
  raw points mode		no
  tile path			no
  gb12345 encoding		no
  fontforge-extras	yes	no
  potrace or autotrace	no
```
How abilitate those features?

**Comment by skef** [comment]
Keep in mind that you don't necessarily *want* every feature that can be added.

From reading through `configure`:

* raw points mode: `--enable-debug-raw-points`. 
* tile path: `--enable-tile-path`
* gb12345: `--enable-gb12345` (but you might run into #1394 issues)
* fontforge-extras: `--enable-fontforge-extras`

**Comment by samiel60** [comment]
ok. With gb12345 compilation fails. Without that option it works fine.
A last question: because I receive 
Optional Library	UseIt?	HaveIt?	WebsiteURL
woff2			no		https://github.com/google/woff2
if I've woff2 installed? A further option to pass?

**Comment by ctrlcctrlv** [comment]
`--enable-woff2`

**Comment by samiel60** [comment]
Now with this configuration:
` ./configure --with-freetype-source=/home/samiel/freetype-2.9.1/ --enable-debug-raw-points --enable-tile-path --enable-fontforge-extras --enable-woff2`
I receive
```
tilepath.c: In function ‘CVTile’:
tilepath.c:1310:5: error: implicit declaration of function ‘CVPreserveState’; did you mean ‘CVPreserveTState’? [-Werror=implicit-function-declaration]
     CVPreserveState((CharViewBase *) cv);
     ^~~~~~~~~~~~~~~
     CVPreserveTState
tilepath.c: In function ‘SCTile’:
tilepath.c:1326:5: error: implicit declaration of function ‘SCPreserveLayer’; did you mean ‘SCClearLayer’? [-Werror=implicit-function-declaration]
     SCPreserveLayer(sc,layer,false);
     ^~~~~~~~~~~~~~~
     SCClearLayer
tilepath.c: In function ‘SCTilePattern’:
tilepath.c:1410:12: error: implicit declaration of function ‘RefCharsCopy’; did you mean ‘RefCharsFree’? [-Werror=implicit-function-declaration]
      ref = RefCharsCopy(pattern->refs);
            ^~~~~~~~~~~~
            RefCharsFree
tilepath.c:1410:10: error: assignment makes pointer from integer without a cast [-Werror=int-conversion]
      ref = RefCharsCopy(pattern->refs);
          ^
tilepath.c: In function ‘PTD_OK’:
tilepath.c:1638:20: error: assignment makes pointer from integer without a cast [-Werror=int-conversion]
  td->pattern->refs = RefCharsCopy(ptd->sc_first.layers[ly_fore].refs);
                    ^
tilepath.c: In function ‘CVPatternTile’:
tilepath.c:1954:28: error: implicit declaration of function ‘CVLayer’; did you mean ‘CVSetLayer’? [-Werror=implicit-function-declaration]
     SCTilePattern(cv->b.sc,CVLayer((CharViewBase *) cv),td.pattern, &td.patternSize, &td.repeatCnt);
                            ^~~~~~~
                            CVSetLayer
cc1: some warnings being treated as errors
Makefile:2712: recipe for target 'libfontforgeexe_la-tilepath.lo' failed
make[3]: *** [libfontforgeexe_la-tilepath.lo] Error 1
make[3]: uscita dalla directory "/home/samiel/src/github.com/fontforge/fontforge/fontforgeexe"
Makefile:2889: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
make[2]: uscita dalla directory "/home/samiel/src/github.com/fontforge/fontforge/fontforgeexe"
Makefile:1946: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: uscita dalla directory "/home/samiel/src/github.com/fontforge/fontforge"
Makefile:1783: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by samiel60** [comment]
Moreover:
```
tilepath.c: In function ‘CVTile’:
tilepath.c:1310:5: error: implicit declaration of function ‘CVPreserveState’; did you mean ‘CVPreserveTState’? [-Werror=implicit-function-declaration]
     CVPreserveState((CharViewBase *) cv);
     ^~~~~~~~~~~~~~~
     CVPreserveTState
tilepath.c: In function ‘SCTile’:
tilepath.c:1326:5: error: implicit declaration of function ‘SCPreserveLayer’; did you mean ‘SCClearLayer’? [-Werror=implicit-function-declaration]
     SCPreserveLayer(sc,layer,false);
     ^~~~~~~~~~~~~~~
     SCClearLayer
tilepath.c: In function ‘SCTilePattern’:
tilepath.c:1410:12: error: implicit declaration of function ‘RefCharsCopy’; did you mean ‘RefCharsFree’? [-Werror=implicit-function-declaration]
      ref = RefCharsCopy(pattern->refs);
            ^~~~~~~~~~~~
            RefCharsFree
tilepath.c:1410:10: error: assignment makes pointer from integer without a cast [-Werror=int-conversion]
      ref = RefCharsCopy(pattern->refs);
          ^
tilepath.c: In function ‘PTD_OK’:
tilepath.c:1638:20: error: assignment makes pointer from integer without a cast [-Werror=int-conversion]
  td->pattern->refs = RefCharsCopy(ptd->sc_first.layers[ly_fore].refs);
                    ^
tilepath.c: In function ‘CVPatternTile’:
tilepath.c:1954:28: error: implicit declaration of function ‘CVLayer’; did you mean ‘CVSetLayer’? [-Werror=implicit-function-declaration]
     SCTilePattern(cv->b.sc,CVLayer((CharViewBase *) cv),td.pattern, &td.patternSize, &td.repeatCnt);
                            ^~~~~~~
```

**Comment by skef** [comment]
Skip tile-path for now; you can add it back in once the fix is merged into the master.

**Comment by ctrlcctrlv** [comment]
Why do you even want to compile with every conceivable feature enabled? What does tile path even do?

**Comment by jtanx** [comment]
gb12345 being broken is known, there's probably a ticket of that somewhere

**Comment by samiel60** [comment]
I tried various options only for testing. Without both  `--enable-tile-path` and `--enable-gb12345` it compiles well. I realized that `--enable-gb12345` is broken. There is an incompatibility, al least in my system, between `--enable-tile-path` and `--enable-woff2`
